src/js/form.js

```diff
@@ -313,26 +328,32 @@ Form.prototype.init = function () {
 
         // after radio button data-name setting (now done in XLST)
         // Set temporary event handler to ensure calculations in newly added repeats are run for the first time
-        const tempHandlerAddRepeat = (event) => this.calc.update(event.detail);
-        const tempHandlerRemoveRepeat = () => (this.all = {});
```

Comment:
I don't immediately see if your changes eliminated the need for this remove handler, or that there wasn't a need in the first place. I'm guessing this removed remove handler was for a form with a default repeat (or several defaults which is possible in XForm, though not in XLSForm) but with a repeat-count of 0. I just wanted to flag this in case there isn't a test for this.

Reply:
I kept what I believe to be the same logic, it just also moved to after repeat initialization [here](https://github.com/enketo/enketo-core/pull/881/files/5e9c82fc29cc78ca1c9b79423d5a24b94a669800#diff-608210f16bb20aa58380a2606ca9a842e85e4634d9e3d03347845848b9bc0f8cR351). I will double check, but I'm reasonably certain that a test with a repeat count of 0 did fail without that line.

Reply:
Ah, I see. I guess they are added first and then removed. I'll also play around with this in the new PR.